src/Mvc/Mvc.Core/src/ProducesResponseTypeAttribute.cs

```diff
@@ -33,6 +34,37 @@ public ProducesResponseTypeAttribute(Type type, int statusCode)
             Type = type ?? throw new ArgumentNullException(nameof(type));
             StatusCode = statusCode;
             IsResponseTypeSetByDefault = false;
+            ContentTypes = new();
+        }
+
+        /// <summary>
+        /// Initializes an instance of <see cref="ProducesResponseTypeAttribute"/>.
+        /// </summary>
+        /// <param name="type">The <see cref="Type"/> of object that is going to be written in the response.</param>
+        /// <param name="statusCode">The HTTP response status code.</param>
+        /// <param name="contentType">The content type associated with the response.</param>
+        /// <param name="additionalContentTypes">Additional content types supported by the response.</param>
+        public ProducesResponseTypeAttribute(Type type, int statusCode, string? contentType, params string[] additionalContentTypes)
+        {
+            Type = type ?? throw new ArgumentNullException(nameof(type));
+            StatusCode = statusCode;
+            IsResponseTypeSetByDefault = false;
+
+            if (!string.IsNullOrEmpty(contentType))
+            {
+                MediaTypeHeaderValue.Parse(contentType);
+                for (var i = 0; i < additionalContentTypes.Length; i++)
```

Comment:
Null check? Though you'd have to _really_ be doing something obtuse to pass in a null.

Reply:
Yeah, I don't think a null-check will provide much value here for the 98% scenarios.